README.md

```diff
@@ -4,4 +4,63 @@
+
+## External Resources
+
+- [Functional Programming with Bananas, Lenses, Envelopes and Barbed Wire](http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.41.125) – the iconic paper that collected a lot of this info for the first time
+- [Recursion Schemes: A Field Guide (Redux)](http://comonad.com/reader/2009/recursion-schemes/) – Ed Kmett’s summary (with links to Haskell code) of various folds and unfolds
+- [Unifying Structured Recursion Schemes](http://www.cs.ox.ac.uk/people/jeremy.gibbons/publications/urs.pdf) – a newer paper on how to generalize recursion schemes
+- [Efficient Nanopass Compilers using Cats and Matryoshka](https://github.com/sellout/recursion-scheme-talk/blob/master/nanopass-compiler-talk.org) – Greg Pfeil’s (Scala) talk on this library (and some other things)
+- [Fix Haskell (by eliminating recursion)](https://github.com/sellout/recursion-scheme-talk/blob/master/recursion-scheme-talk.org) – Greg Pfeil’s (Haskell) talk on recursion schemes
+
+## Introduction
+
+This library is predicated on the idea of rewriting your recursive data structures as functors where the recursive component is replaced by a parameter.
+
+```scala
+sealed trait Expr
+final case class Var(name: String)          extends Expr
+final case class Lam(v: String, body, Expr) extends Expr
+final case class App(f: Expr, arg: Expr)    extends Expr
+```
+could be rewritten as
+
+```scala
+sealed trait Expr[A]
+final case class Var[A](name: String)       extends Expr[A]
+final case class Lam[A](v: String, body, A) extends Expr[A]
+final case class App[A](f: A, arg: A)       extends Expr[A]
+```
+
+Then you use one of the fixed point type constructors below to regain your recursive type.
+
+### Fixpoint Types
+
+These types take a one-arg type constructor (a functor) and provide a recursive form of it.
+
+All of these types have instances for `Recursive`, `Corecursive`, `FunctorT`, `TraverseT`, `Equal`, `Show`, and `Arbitrary` unless otherwise noted.
+
+- `Fix` – This is the simplest fixpoint type, implemented with general recursion.
+- `Mu` – This is for inductive (finite) recursive structures, models the concept of “data”, aka, the “least fixed point”.
+- `Nu` – This is for coinductive (potentially infinite) recursive structures, models the concept of “codata”, aka, the “greatest fixed point’.
+- `Cofree[?[_], A]` – Does not have a `Corecursive` instance. This represents a structure with some metadata attached to each node. In addition to the usual operations, it can also be folded using an Elgot algebra.
+- `Free[?[_], A]` – Does not have a `Recursive` instance. In addition to the usual operations, it can also be created by unfolding with an Elgot coalgebra.
+
+### Algebras
+
+A structure like this makes it possible to separate recursion from your operations. You can now write transformations that operate on only a single node of your structure at a time.
+
```

Comment:
Again, a really small example would help tremendously here. Probably a `cata` over some small Expr, but I'm having trouble thinking of one that's very obvious. It would be easier if you had gone with the usual `Expr = Num Integer | Mul Expr Expr` and `eval`